convert_hf_to_gguf.py

```diff
@@ -417,15 +418,13 @@ def get_model_part_names(dir_model: Path, prefix: str, suffix: str) -> list[str]
 
     @staticmethod
     def load_hparams(dir_model: Path):
-        with open(dir_model / "config.json", "r", encoding="utf-8") as f:
-            hparams = json.load(f)
-            architectures = hparams.get("architectures")
-            if "text_config" in hparams:
-                hparams = {**hparams, **hparams["text_config"]}
-            if architectures is not None:
-                # preserve "architectures" from root level config
-                hparams["architectures"] = architectures
-            return hparams
+        try:
+            return AutoConfig.from_pretrained(dir_model, trust_remote_code=True).to_dict()
```

Comment:
```suggestion
            return AutoConfig.from_pretrained(dir_model, trust_remote_code=False).to_dict()
```

I don't think `trust_remote_code=True` should be the default here.

If a model uses a custom module, then hopefully it also has the relevant information in `config.json` (and/or we can assume some defaults as usual).

Reply:
The reason why it left that `trust_remote_code=True` was because we already had it in some places in the code, mostly to load the tokenizer. But on second thought, yeah this can be a huge security risk. Basically any bad actors can just trick user to try their model, and they can add a command execution inside.

I think the fallback to `config.json` should work well for now, given that not many models need `trust_remote_code`

One thing I'm a bit concern though, should we guard all other places with `trust_remote_code=True` behind a flag, like for example `--trust-remote-code`? This is good for performance, but can be quite a bad UX for people don't know why it's needed